test/electron-renderer/ui/modals/GoToModal_spec.js

```diff
@@ -74,6 +79,77 @@ createModalTest(GoToModal, ['gotourl'], [], [], () => {}, (_c) => {
     fired.generateDebugInfo.should.be.ok;
   });
 
+  describe('stubbed fetch request', () => {
+    beforeEach(() => {
+      sinon.stub(window, 'fetch');
+
+      const fakeRedirect = () =>
+        Promise.resolve({
+          url: 'http://google.com/redirected',
+          status: 301,
+        });
+
+      const fakeOk = () =>
+        Promise.resolve({
+          url: 'https://play.google.com/music/listen#/album/Brktxc3e6ajhek4mxpmaa3quksu',
+          status: 200,
+        });
+
+      // setup mocks for fetch functionality (like redirects)
+      window.fetch.onFirstCall().returns(fakeRedirect());
+      window.fetch.onSecondCall().returns(fakeRedirect());
+      window.fetch.onThirdCall().returns(fakeOk());
+    });
+
+    afterEach(() => {
+      window.fetch.restore();
+    });
+
+    given(['https://goo.gl/UFT1HU', 'https://goo.gl/ySV9WV'])
+    .it('should emit the navigate event if it is a valid redirected GPM URL', (url) => {
```

Comment:
This indenting looks weird, probably better to put it all on one line and if required ignore the eslint rule on this line